### PR TITLE
Factory bot Update

### PR DESCRIPTION
**Pull request by zurbergram**
## Description
https://github.com/department-of-veterans-affairs/vets.gov-team/issues/19788

https://thoughtbot.com/blog/deprecating-static-attributes-in-factory_bot-4-11

As a developer, I need to update the factory_bot to the latest version so that GIDS is in sync with the other repositories.

## Assumptions:
1. Similar work has been done in vets-api:
     - https://github.com/department-of-veterans-affairs/vets-api/pull/3300

## Acceptance Criteria:
1.  Rspec build passes with new versions of the dependency
2.  GIDS UI is displayed properly
3.  All unit tests pass

## Supporting Artifacts:
1. Potential fixes to factories: https://github.com/thoughtbot/factory_bot/commit/50eeb67241ea78a6b138eea694a2a25413052f49

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

File: spec/models/storage_spec.rb [hunk 1]
```diff
@@ -11,7 +11,13 @@ def generate_csv_upload(name)
...
   end
 
-  subject { build :storage }
+  subject { build :storage, user: user }
+
+  let(:user) { User.first }
+
+  before do
+    create :user, email: 'fred@va.gov', password: 'fuggedabodit'
+  end
```
Comment (omgitsbillryan) [inline spec/models/storage_spec.rb]: Can you describe to me what's going on here? Even after reading the thoughtbot article in your OP, I don't quite follow.

Comment (jholton) [inline spec/models/storage_spec.rb]: Can we reduce these lines to this?
```
subject { build :storage, user: user }

let(:user) { create :user, email: 'fred@va.gov', password: 'fuggedabodit' }
```

File: spec/models/upload_spec.rb [hunk 2]
```diff
@@ -3,7 +3,13 @@
...
 
 RSpec.describe Upload, type: :model do
-  subject { build :upload }
+  subject { build :upload, user: user }
+
+  let(:user) { User.first }
+
+  before do
+    create :user, email: 'fred@va.gov', password: 'fuggedabodit'
+  end
```
Comment (jholton) [inline spec/models/upload_spec.rb]: Same as above, can we reduce to this?
```
subject { build :upload, user: user }

let(:user) { create :user, email: 'fred@va.gov', password: 'fuggedabodit' }
```

File: spec/models/version_spec.rb [hunk 3]
```diff
@@ -3,8 +3,14 @@
 require 'rails_helper'
 
 RSpec.describe Version, type: :model do
+  let(:user) { User.first }
+
+  before do
+    create :user, email: 'fred@va.gov', password: 'fuggedabodit'
+  end
```
Comment (jholton) [inline spec/models/version_spec.rb]: Same reduction recommendation:
`let(:user) { create :user, email: 'fred@va.gov', password: 'fuggedabodit' }`

File: spec/models/storage_spec.rb (hunk 1 above)
Comment (zurbergram) [reply spec/models/storage_spec.rb]: When I updated Factorybot the static attributes in the various factories were causing errors when running rspec.  The article mentions that a cop was created for Rubocop to easily fix this issue, in https://github.com/department-of-veterans-affairs/gibct-data-service/pull/467 Rubocop was updated to gain access to the new cop and auto-correct all factories.

Comment (zurbergram) [reply spec/models/storage_spec.rb]: For these specific changes I'm not sure why they broke but the issue seems to be with how the object is constructed and references an existing User object on creation, errors for things changed were around the User of the model being nil so I set the user property in the tests that were failing

File: spec/models/version_spec.rb (hunk 3 above)
Comment (omgitsbillryan) [reply spec/models/version_spec.rb]: ➕

File: spec/models/storage_spec.rb (hunk 1 above)
Comment (omgitsbillryan) [reply spec/models/storage_spec.rb]: ➕

Comment (omgitsbillryan) [reply spec/models/storage_spec.rb]: Gotcha, I get it now. J/c, why are you overriding the `email` and `password` attributes over using the `Faker` attrs that the users factory uses?

**Comment by annaswims** [comment]
please use words in the body of the pull request to describe what you've done.  don't just link to an issue.  It will help us context switch and review pr's more quickly.

File: spec/models/storage_spec.rb (hunk 1 above)
Comment (zurbergram) [reply spec/models/storage_spec.rb]: copy paste from another spec, removed

**Review by omgitsbillryan — APPROVED** [review]

**Pull request merged**